src/LinearMaps.jl

```diff
@@ -25,16 +25,22 @@ function LinearAlgebra.mul!(y::AbstractVector, A::LinearMap, x::AbstractVector)
     length(y) == size(A, 1) || throw(DimensionMismatch("mul!"))
     A_mul_B!(y, A, x)
 end
-
 # the following is of interested in, e.g., subspace-iterative methods
 function LinearAlgebra.mul!(Y::AbstractMatrix, A::LinearMap, X::AbstractMatrix)
     (size(Y, 1) == size(A, 1) && size(X, 1) == size(A, 2) && size(Y, 2) == size(X, 2)) || throw(DimensionMismatch("mul!"))
-    Y .= Matrix(A * X)
+    @inbounds @views for i=1:size(X, 2)
+        mul!(Y[:, i], A, X[:, i])
+    end
     return Y
 end
 # the following is required for TSVD.jl
-function LinearAlgebra.mul!(α::Number, A::LinearMap, x::Vector, β::Number, y::Vector)
-    y .= α * A * x + β * y
+function LinearAlgebra.mul!(α::Number, A::LinearMap, x::AbstractVector, β::Number, y::AbstractVector)
+    if β != one(β)
```

Comment:
The only precedent for a 5-argument `mul!` in Base/Stdlib (which I would very much like to see) seems to be for Sparse arrays (see https://discourse.julialang.org/t/why-doesnt-mul-have-a-gemv-style-interface/12136 ). It seems that, with Julia 0.7 and the transition from `A(x)_mul_B!` to `mul!`, the argument order has changed in a good way, no longer following BLAS but such that 3-arg mul is like 5-arg mul with default values for arguments: `mul!(C, A, B, alpha = 1, beta = 0)`. So I hope that `TSVD.jl` will follow that convention.

Reply:
Thanks @Jutho for your comments and the link. There were so many instances of `mul!` or their predecessors in Julia 0.6 that it was hard to find the 5-arg versions in the first place. As for Julia 0.7, I don't have Juno working with it, and I had no idea how to crawl the docs for the right argument signature (the julialang.org docs don't mention the 5-arg version). If there is a `mul!` with the `C, A, B` signature, I'll adopt to that. I'm sure Andreas Noack is aware of that convention and will follow. I'll file an issue at `TSVD.jl` to track any changes in a Julia 0.7 update. After that, I'd just merge (will need to ask my colleague about how to tag a new version etc.). Since this is an extension without breaking anything, this should be then version 2.0.1?